packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -69,6 +68,72 @@ const safeGetCache = ({ getCache, cache }) => id => {
   return getCache(id)
 }
 
+/**
+ * @template T
+ * @param {Array<T>} input
+ * @param {(input: T) => Promise<void>} iterator
+ * @return Promise<void>
+ */
+const eachPromise = (input, iterator) =>
+  input.reduce(
+    (accumulatorPromise, nextValue) =>
+      accumulatorPromise.then(() => void iterator(nextValue)),
```

Comment:
this `void` here is problematic? after removing it, it seems to work - why is it here? some context is needed

Reply:
Also generally - where do we stand on this? We use bluebird pretty heavily--so I'm not _sure_ it was worth refactoring this.

But a bit tangential, if we went that way--just need to ensure we have a working alternative for sequential promises!

Reply:
I personally don't agree with removing dependency for the sake of removing it. If it fixes bugs or improve performance, then that's fine.

I will open PR fixing that and add some tests for this helper.

Reply:
When utils like `bluebird` have maintainers with domain knowledge and it was battle-tested by thousands of developers - I will trust that more than hand rolled and not very tested approaches.

Reply:
In any case - here's hot-fix for issue https://github.com/gatsbyjs/gatsby/pull/12578 (first commit on purpose just added failing test, and second commit "fixes" the problem.

@alexkirsz I would appreciate you taking a look on this, because I imagine there was some reason for that `void` to be there, but I have no clue what that would be (and it did break that utility)

Reply:
@pieh I'm not the author of this particular piece of code, @wardpeet is (I only authored the first commit of this PR).

Reply:
Ah, ok - sorry @alexkirsz. I should pay more attention to individual commits in the PR

Reply:
my bady, sorry about this